src/modules/universal-discovery/components/finder/finder.component.js

```diff
@@ -18,39 +19,142 @@ export default class FinderComponent extends Component {
         this.appendMoreItems = this.appendMoreItems.bind(this);
         this.updateLocationsData = this.updateLocationsData.bind(this);
         this.findLocationChildren = this.findLocationChildren.bind(this);
+        this.loadBranchChildren = this.loadBranchChildren.bind(this);
         this.onLoadMore = this.onLoadMore.bind(this);
         this.renderBranch = this.renderBranch.bind(this);
+
+        this.locationsMap = {};
+        this.activeLocations = {};
+        this.preselectedItem = null;
     }
 
     componentDidMount() {
-        this.props.findLocationsByParentLocationId(
-            Object.assign({}, this.props.restInfo, { parentLocationId: this.props.startingLocationId}),
-            this.updateLocationsData
-        );
+        const isForcedLocation = this.props.allowedLocations.length === 1;
+        const allowedIncludesPreselected = this.props.allowedLocations.includes(this.props.preselectedLocation);
+        const isPreselectedLocation = this.props.preselectedLocation && (!this.props.allowedLocations.length || allowedIncludesPreselected);
+
+        if (isForcedLocation) {
+            this.loadPreselectedData(this.props.allowedLocations[0], true);
+        } else if (isPreselectedLocation) {
+            this.loadPreselectedData(this.props.preselectedLocation, true);
+        } else {
+            this.props.findLocationsByParentLocationId(
+                Object.assign({}, this.props.restInfo, { parentLocationId: this.props.startingLocationId }),
+                this.updateLocationsData
+            );
+        }
     }
 
     componentDidUpdate() {
         this.updateBranchesContainerScroll();
     }
 
+    componentWillReceiveProps(props) {
+        if (this.preselectedItem && this.locationsMap && this.activeLocations && props.isVisible && props.isVisible !== this.props.isVisible) {
+            this.setState(state => Object.assign({}, state, {
+                locationsMap: this.locationsMap,
+                activeLocations: this.activeLocations
+            }), () => this.props.isVisible && this.props.onItemSelect(this.preselectedItem));
+        }
+    }
+
+    /**
+     * Load data of preselected location.
+     *
+     * @method loadPreselectedData
+     * @param {String} locationId
+     * @memberof FinderComponent
+     */
+    loadPreselectedData(id) {
+        const promise = new Promise(resolve => loadAccordionData(
+            {startingLocationId: this.props.startingLocationId, locationId: id},
+            resolve
+        ));
+
+        promise
+            .then((response) => {
+                this.createAccordionData(response);
```

Comment:
You're creating so called accordion data here, but not reusing it in `this.setAccordionData()` is that correct?